runtime/protocol/http-client-engines/http-client-engine-crt/common/src/aws/smithy/kotlin/runtime/http/engine/crt/SdkStreamResponseHandler.kt

```diff
@@ -172,9 +175,10 @@ internal class SdkStreamResponseHandler(
         // close the body channel
         if (errorCode != 0) {
             val errorDescription = CRT.errorString(errorCode)
-            val ex = CancellationException("CrtHttpEngine::response failed: ec=$errorCode; description=$errorDescription")
+            val errName = CRT.errorName(errorCode)
+            val ex = HttpException("$errName: $errorDescription; ec=$errorCode", errorCode = mapCrtErrorCode(errorCode))
             responseReady.close(ex)
-            bodyChan.cancel(ex)
+            bodyChan.close(ex)
```

Comment:
`cancel` requires a `CancellationException` and generally would be used to indicate the producer is cancelled whereas `close` (with exception) indicates the channel is failed. Using `close` is more appropriate here and will result in the consumer seeing an `HttpException` rather than a wrapped exception.